### ISSUE TITLE
Use MongoDB as datasource in a multiple-datasource configuration

### ISSUE THREAD

**Issue by sojaducana** [robertoschwald/grails-audit-logging-plugin#93]
Using v1.0.5 of the plugin and MongoDB plugin v3.0.3, how can we set the plugin to use MongoDB as the datasource?

Here's a relevant snippet in DataSource.config:
```
dataSource {
    pooled = true
    jmxExport = true
    dbCreate = "update"
    driverClassName = "com.mysql.jdbc.Driver"
}

grails {
    mongo {
        host = "localhost"
        port = 27017
        databaseName = "mongoDb"
    }
}
```

I tried using useDatasource = 'mongo', unfortunately that doesn't seem to work.

One of the things I tried to make it work was to explicitly declare in AuditLogEvent:
```
static mapWith = "mongo"
```

Changed mapping to:
```
collection Holders.config.auditLog.tablename ?: 'audit_log'
```

And removed from mapping:
```
datasource "$Holders.config.auditLog.useDatasource"
```

Is it possible for this kind of support to be added in a future release?

Thanks.

**Comment by robertoschwald** [comment]
In the upcoming Grails 3.0 Audit-Logging Plugin 2.0.0 version, we plan to hand over full control to the AuditLogEvent domain class to the plugin user (see #13 ). This way, you can add whatever mapping you need. It might be possible that we backport this feature to the 1.x versions for usage in Grails < 3.0.

**Comment by bronoman** [comment]
Can you include an example how to manage a mongoDB for this part of the documentation:
> #### Use specific datasource (since 1.0.4)
> Since version 1.0.4 it is possible to set the datasource name to use for the audit logging table.
> ```useDatasource = 'second' // use "second" datasource defined in DataSource.groovy```

...for the example above, what should be the correct value to be passed: 'second', 'grails', 'mongo', 'localhost' or 'mongoDB' ?

**Comment by robertoschwald** [comment]
Unfortunately, the mongodb plugin does not support the default datasource mapping like:

   static mapping = {
      datasource second
   }

but uses the "mapWith" static property, which normally is only used to control persistence of a domain instance in Grails (http://grails.github.io/grails-doc/2.4.4/ref/Domain%20Classes/mapWith.html).

Does anyone have an idea how we could make that static field configurable until we support #13 ?

**Comment by robertoschwald** [comment]
See also #99 

Is fixed in latest 1.0.6-SNAPSHOT. Tested with Grails 2.4.4, mongodb plugin 3.0.3 as only DS in a sample application. MongoDB server 3.0.5 on OSX.

Problem was, that the mongodb datastore does not hold a config property.

Please try.

**Comment by robertoschwald** [comment]
Published 1.1.0-SNAPSHOT with externalited AuditLogEvent domain class (see #13 ). Would be great if you can test it.

**Comment by Beyek** [comment]
@robertoschwald 
I integrated this plug-in (version 2.0.6) with my Grails 3.1.4 plugin (which is a generalized plugins for different grails application) and I am facing similar problem(Use secondaryDB as datasource in a multiple-datasource configuration) . The only difference is the database is mysql db.

The datasource config is as follows.
dataSources:
    dataSource:
        pooled: true
        jmxExport: true
        driverClassName: com.mysql.jdbc.Driver
        dialect: "org.hibernate.dialect.MySQL5InnoDBDialect"

    usermanagement:
        pooled: true
        jmxExport: true
        driverClassName: com.mysql.jdbc.Driver
        dialect: "org.hibernate.dialect.MySQL5InnoDBDialect"
environments:
    development:
        dataSources:
            dataSource:
                dbCreate: update
                url: jdbc:mysql://localhost:3306/primaryDB?autoReconnect=true&zeroDateTimeBehavior=convertToNull
              
            secondaryDB:
                dbCreate: update
                url: jdbc:mysql://localhost:3306/secondaryDB?zeroDateTimeBehavior=convertToNull
       

My AuditLogEvent,groovy  has mapping as 
###     static mapping = {

        // Set similiar when you used "auditLog.tablename" in < 1.1.0 plugin version.
        table 'audit_log'

        // Remove when you used "auditLog.cacheDisabled = true" in < 1.1.0 plugin version.
        cache usage: 'read-only', include: 'non-lazy'

        // Set similiar when you used "auditLog.useDatasource" in < 1.1.0 plugin version.
         datasource "secondaryDB"

        // Set similiar when you used "auditLog.idMapping" in < 1.1.0 plugin version. Example:
        // id generator:"uuid2", type:"string", "length:36"

        // no HQL queries package name import (was default in 1.x version)
        //autoImport false

        version false
    }

When I define the datasource there is a compilation error. 

> ### Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'transactionManager': Cannot resolve reference to bean 'transactionManager_secondaryDb' while setting constructor argument with key [3]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'transactionManager_secondaryDb': Cannot resolve reference to bean 'sessionFactory_secondaryDb' while setting bean property 'sessionFactory'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'sessionFactory_secondaryDb': Invocation of init method failed; nested exception is org.hibernate.cache.NoCacheRegionFactoryAvailableException: Second-level cache is used in the application, but property hibernate.cache.region.factory_class is not given; please either disable second level cache or set correct region factory using the hibernate.cache.region.factory_class setting and make sure the second level cache provider (hibernate-infinispan, e.g.) is available on the classpath.

But when I remove the datasource definition, the Application successfully compiles but the audit_log table is created in primaryDB. 
The audit_log table should be in secondaryDB rather than in primaryDB, as this should be implement in custom plugin.
I have followed the guideline mentioned [https://github.com/robertoschwald/grails-audit-logging-plugin](url)
What did I missed here, could you please suggest?
